### PR TITLE
Fix error message on expression evaluation  for module load error

### PR DESCRIPTION
**Pull request by annagrin**
Change error message on expression evaluation using deferred libraries from `NetworkError` to `LoadModuleError` with more readable details about the module name and path.

Closes: https://github.com/dart-lang/webdev/issues/1541

### CODE REVIEW

File: dwds/lib/src/services/expression_evaluator.dart [hunk 1]
```diff
@@ -42,8 +45,11 @@ class ExpressionEvaluator {
   static final _syntheticNameFilterRegex =
       RegExp('org-dartlang-debug:synthetic_debug_expression:.*:.*Error: ');
 
-  ExpressionEvaluator(
-      this._inspector, this._locations, this._modules, this._compiler);
+  static final _modulePathRegex =
```
Comment (elliette) [inline dwds/lib/src/services/expression_evaluator.dart:48]: A few suggestions: it looks like this only gets used in `_formatEvaluationError`, does it make sense to just declare it there? Also might be better to rename to `_moduleLoadErrorRegex` or something like that to make it clearer.

File: dwds/lib/src/services/expression_evaluator.dart [hunk 2]
```diff
@@ -42,8 +45,11 @@ class ExpressionEvaluator {
...
+      RegExp(r".*Failed to load '.*\.com/(.*\.js).*");
```
Comment (elliette) [inline dwds/lib/src/services/expression_evaluator.dart:49]: Could we also add a comment with an example of an error that this will match on?

File: dwds/lib/src/services/expression_evaluator.dart [hunk 3]
```diff
@@ -240,16 +246,26 @@ class ExpressionEvaluator {
...
+      } else if (error.startsWith('NetworkError: ')) {
+        var modulePath = _modulePathRegex.firstMatch(error)?.group(1);
+        var module = modulePath != null
+            ? await globalLoadStrategy.moduleForServerPath(
+                _entrypoint, modulePath)
+            : 'unknown';
+        modulePath ??= 'unknown';
+        error = 'Module is not loaded : $module (path: $modulePath). '
+            'Deferred library loading in expression '
+            'evaluation is not supported.';
```
Comment (nshahan) [inline dwds/lib/src/services/expression_evaluator.dart:267]: Maybe rephrase this since it isn't likely that someone was trying to load a library. It's more likely that it was a side effect of some expression since JIT modules can cause this issue without any deferred imports.

Something like "Accessing libraries that have not yet been used in the application is not supported during expression evaluation." ?

**Review by nshahan — APPROVED** [review]

File: dwds/lib/src/services/expression_evaluator.dart (hunk 1 above)
Comment (annagrin) [reply dwds/lib/src/services/expression_evaluator.dart:48]: Moving the regex to the function this would create a new object every time we call the function. But it might me a small price to pay, not really sure. I prefer to keep it as is if this is ok. Will change the name!

File: dwds/lib/src/services/expression_evaluator.dart (hunk 2 above)
Comment (annagrin) [reply dwds/lib/src/services/expression_evaluator.dart:49]: sure, will do!

File: dwds/lib/src/services/expression_evaluator.dart (hunk 3 above)
Comment (annagrin) [reply dwds/lib/src/services/expression_evaluator.dart:267]: sounds good!

Comment (annagrin) [reply dwds/lib/src/services/expression_evaluator.dart]: Done!

File: dwds/lib/src/services/expression_evaluator.dart (hunk 2 above)
Comment (annagrin) [reply dwds/lib/src/services/expression_evaluator.dart:54]: done!

File: dwds/lib/src/services/expression_evaluator.dart (hunk 1 above)
Comment (annagrin) [reply dwds/lib/src/services/expression_evaluator.dart]: Changed the name!

**Review by elliette — APPROVED** [review]
